sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs

```diff
@@ -369,4 +410,304 @@ public static partial class BlobsModelFactory
+
+        /// <summary>
+        /// Returns the name of the encryption scope used to encrypt the blob contents and application metadata.  Note that the absence of this header implies use of the default account encryption scope.
+        /// </summary>
+        public string EncryptionScope { get; internal set; }
+
+        /// <summary>
+        /// The tier of page blob on a premium storage account or tier of block blob on blob storage LRS accounts. For a list of allowed premium page blob tiers, see https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#features. For blob storage LRS accounts, valid values are Hot/Cool/Archive.
+        /// </summary>
+        public string AccessTier { get; internal set; }
+
+        /// <summary>
+        /// For page blobs on a premium storage account only. If the access tier is not explicitly set on the blob, the tier is inferred based on its content length and this header will be returned with true value.
+        /// </summary>
+        public bool AccessTierInferred { get; internal set; }
+
+        /// <summary>
+        /// For blob storage LRS accounts, valid values are rehydrate-pending-to-hot/rehydrate-pending-to-cool. If the blob is being rehydrated and is not complete then this header is returned indicating that rehydrate is pending and also tells the destination tier.
+        /// </summary>
+        public string ArchiveStatus { get; internal set; }
+
+        /// <summary>
+        /// The time the tier was changed on the object. This is only returned if the tier on the block blob was ever set.
+        /// </summary>
+        public DateTimeOffset AccessTierChangedOn { get; internal set; }
+
+        /// <summary>
+        /// A DateTime value returned by the service that uniquely identifies the blob. The value of this header indicates the blob version, and may be used in subsequent requests to access this version of the blob.
+        /// </summary>
+        public string VersionId { get; internal set; }
+
+        /// <summary>
+        /// The value of this header indicates whether version of this blob is a current version, see also x-ms-version-id header.
+        /// </summary>
+        public bool IsCurrentVersion { get; internal set; }
+
+        /// <summary>
+        /// The number of tags associated with the blob
+        /// </summary>
+        public long TagCount { get; internal set; }
+
+        /// <summary>
+        /// The time this blob will expire.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; internal set; }
+
+        /// <summary>
+        /// If this blob has been sealed
+        /// </summary>
+        public bool IsSealed { get; internal set; }
+
+        /// <summary>
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID.
+        /// </summary>
+        /// <returns></returns>
+        internal IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds()
```

Comment:
Is this the same logic as `BlobDownloadDetails.ParseObjectReplicationIds()`?  If so, I think we should consolidate it and move it to `BlobExtensions`.